packages/vite-environment-provider-node-vm/src/index.ts

```diff
@@ -151,7 +152,7 @@ async function createNodeVmDevEnvironment(
                   });
                 },
                 send(message) {
-                  eventEmitter.emit("message",JSON.stringify(message));
+                  eventEmitter.emit("message", message);
```

Comment:
The `message` is already a string so this was being serialised twice before. This meant that on the other side `JSON.parse` was returning a string.